File: Infinity.Bnois/Infinity.Bnois.ApplicationService/Implementation/RoasterListService.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Threading.Tasks;
using Infinity.Bnois.ApplicationService.Interface;
using Infinity.Bnois.Data;

namespace Infinity.Bnois.ApplicationService.Implementation
{
    public class RoasterListService : IRoasterListService
    {
        private readonly IBnoisRepository<Employee> employeeRepository;
        public RoasterListService(IBnoisRepository<Employee> employeeRepository)
        {
            this.employeeRepository = employeeRepository;
        }

        public List<object> GetRoasterListByShipType(int shipType, int coxoStatus)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetCoOpvFfWaitingList] {0},{1} ", shipType, coxoStatus));
            return dataTable.ToJson().ToList();
        }

        //public dynamic GetSeaServices(string pno)
        //{

        //    Dictionary<string, dynamic> keyValues = new Dictionary<string, dynamic>();

        //    foreach (var type in Enum.GetValues((typeof(ShipType))))
        //    {
        //        var results = GetSeaServicesByType(pno, (int)type);

        //        var shipType = String.Empty;
        //        if (results.Count > 0)
        //        {
        //            if ((int)type == (int)ShipType.Small)
        //            {
        //                shipType = "Small";

        //            }
        //            else if ((int)type == (int)ShipType.Medium)
        //            {
        //                shipType = "Medium";
        //            }
        //            else
        //            {
        //                shipType = "Large";
        //            }


        //            keyValues.Add(shipType, results);
        //        }

        //    }
        //    return keyValues;
        //}

        public List<object> GetLargeShipProposedWaitingCoXoList(int shipType, int officeId, int appointment)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetCoXoProposedWaitingOfficerList] {0},{1}, {2}", shipType, officeId, appointment));
            return dataTable.ToJson().ToList();
        }
        public List<object> GetLargeShipCoWaitingList()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetLargeCoPendingList]"));
            return dataTable.ToJson().ToList();
        }
        public List<object> GetLargeShipXoWaitingList()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetLargeXoPendingList]"));
            return dataTable.ToJson().ToList();
        }
        public List<object> GetMediumShipCoWaitingList()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetMediumCoPendingList]"));
            return dataTable.ToJson().ToList();
        }
        
        public List<object> GetSmallShipCoXoWaitingList(int shipType, int coxoStatus, int viewStatus)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetSmallCoOpvFfWaitingList] {0},{1},{2} ", shipType, coxoStatus, viewStatus));
            return dataTable.ToJson().ToList();
        }
        public List<object> GetSmallShipCoXoPendingList()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetSmallCoXoPendingList]"));
            return dataTable.ToJson().ToList();
        }

    }
}

File: Infinity.Bnois.Web/Infinity.Bnois.Web/app/services/roasterListService.js
(function () {
    'use strict';
    angular.module('app').service('roasterListService', ['dataConstants', 'apiHttpService', roasterListService]);

    function roasterListService(dataConstants, apiHttpService) {
        var service = {
            getRoasterListByShipType: getRoasterListByShipType,
            getRoasterListBySmallShip: getRoasterListBySmallShip,
            getLargeShipCoWaitingList: getLargeShipCoWaitingList,
            getLargeShipXoWaitingList: getLargeShipXoWaitingList,
            getMediumShipCoWaitingList: getMediumShipCoWaitingList,
            getSmallShipCoXoWaitingList: getSmallShipCoXoWaitingList,
            getLargeShipProposedWaitingCoXoList: getLargeShipProposedWaitingCoXoList

        };

        return service;

        function getRoasterListByShipType(shipType, coxoStatus) {
            var url = dataConstants.ROASTER_LIST_URL + 'get-roaster-list-by-ship-type?shipType=' + shipType + '&coxoStatus=' + coxoStatus;
            return apiHttpService.GET(url);
        }
        function getRoasterListBySmallShip(shipType, coxoStatus, viewStatus) {
            var url = dataConstants.ROASTER_LIST_URL + 'get-roaster-list-by-small-ship?shipType=' + shipType + '&coxoStatus=' + coxoStatus+ '&viewStatus=' + viewStatus;
            return apiHttpService.GET(url);
        }

        function getLargeShipProposedWaitingCoXoList(shipType, officeId, appointment) {
            var url = dataConstants.ROASTER_LIST_URL + 'get-proposed-waiting-coxo-list?shipType=' + shipType + '&officeId=' + officeId + '&appointment=' + appointment;
            return apiHttpService.GET(url);
        }
        function getLargeShipCoWaitingList() {
            var url = dataConstants.ROASTER_LIST_URL + 'get-large-ship-co-waiting-list';
            return apiHttpService.GET(url);
        }
        function getLargeShipXoWaitingList() {
            var url = dataConstants.ROASTER_LIST_URL + 'get-large-ship-xo-waiting-list';
            return apiHttpService.GET(url);
        }
        function getMediumShipCoWaitingList() {
            var url = dataConstants.ROASTER_LIST_URL + 'get-medium-ship-co-waiting-list';
            return apiHttpService.GET(url);
        }
        function getSmallShipCoXoWaitingList() {
            var url = dataConstants.ROASTER_LIST_URL + 'get-small-ship-coxo-waiting-list';
            return apiHttpService.GET(url);
        }
    }
})();